lib/matplotlib/tests/test_collections.py

```diff
@@ -680,6 +680,16 @@ def test_set_wrong_linestyle():
         c.set_linestyle('fuzzy')
 
 
+@mpl.style.context('default')
+def test_zero_linewidth_scaling():
+    c = Collection()
+    c.set_linestyle('dashed')
+    c.set_linewidth(0)
+
+    # With a zero linewidth, there can be no pattern
+    assert c.get_linestyle() == [(0, None)]
```

Comment:
Note that "solid" results in `['', None]` instead which I believe is incorrect wrt. to the declared type.

We are overall not consistent and clear what `get_linestyle()` means:

- for `Line2D.get_linestyle()` it is always a str, and if given a dash pattern, the result is "--".
- for `Patch.get_linestyle()`, we basically get what we put in str or (unscaled) dash pattern; with the exception of normalizing "no style" ' ', '', 'none' to "None"
- for Collections, we always get a scaled dash pattern.


While thes fix may be ok as a workaround to fix the original issue (and doesn't make the consisteny worse), we should generally agee upon what `get_linestyle()` means - and a scaled dash pattern would be the last of my choices.

Reply:
Actually looking closer, I don't think the legend handler should be getting the scaled pattern - it gets scaled again on the patch.

I'll close this.